modules/wurflRtdProvider.js

```diff
@@ -0,0 +1,233 @@
+
+  bidders.forEach((bidderCode) => {
+    if (bidderCode in authBidders) {
+      // inject WURFL data
+      enrichedBidders.add(bidderCode);
+      const data = bidderData(wjsResponse.WURFL, caps, authBidders[bidderCode]);
+      logger.logMessage(`injecting data for ${bidderCode}: `, data);
+      enrichBidderRequest(reqBidsConfigObj, bidderCode, data);
+      return;
+    }
+    // inject WURFL low entropy data
+    const data = lowEntropyData(wjsResponse.WURFL, wjsResponse.wurfl_pbjs?.low_entropy_caps);
+    logger.logMessage(`injecting low entropy data for ${bidderCode}: `, data);
+    enrichBidderRequest(reqBidsConfigObj, bidderCode, data);
+  });
+}
+
+/**
+ * bidderData returns the WURFL data for a bidder
+ * @param {Object} wurflData WURFL data
+ * @param {Array} caps Capability list
+ * @param {Array} filter Filter list
+ * @returns {Object} Bidder data
+ */
+export const bidderData = (wurflData, caps, filter) => {
+  const data = {};
+  caps.forEach((cap, index) => {
+    if (!filter.includes(index)) {
+      return;
+    }
+    if (cap in wurflData) {
+      data[cap] = wurflData[cap];
+    }
+  });
+  return data;
+}
+
+/**
+ * lowEntropyData returns the WURFL low entropy data
+ * @param {Object} wurflData WURFL data
+ * @param {Array} lowEntropyCaps Low entropy capability list
+ * @returns {Object} Bidder data
+ */
+export const lowEntropyData = (wurflData, lowEntropyCaps) => {
+  const data = {};
+  lowEntropyCaps.forEach((cap, _) => {
+    let value = wurflData[cap];
+    if (cap == 'complete_device_name') {
+      value = value.replace(/Apple (iP(hone|ad|od)).*/, 'Apple iP$2');
+    }
+    data[cap] = value;
+  });
+  return data;
+}
+
+/**
+ * enrichBidderRequest enriches the bidder request with WURFL data
+ * @param {Object} reqBidsConfigObj Bid request configuration object
+ * @param {String} bidderCode Bidder code
+ * @param {Object} wurflData WURFL data
+ */
+export const enrichBidderRequest = (reqBidsConfigObj, bidderCode, wurflData) => {
+  const ortb2data = {
+    'device': {
+      'ext': {
+        'wurfl': wurflData,
+      }
+    },
+  };
+  mergeDeep(reqBidsConfigObj.ortb2Fragments.bidder, { [bidderCode]: ortb2data });
+}
+
+/**
+ * onAuctionEndEvent is called when the auction ends
+ * @param {Object} auctionDetails Auction details
+ * @param {Object} config Configuration for WURFL RTD submodule
+ * @param {Object} userConsent User consent data
+ */
+function onAuctionEndEvent(auctionDetails, config, userConsent) {
+  const altHost = config.params?.altHost ?? null;
+
+  let host = WURFL_JS_HOST;
+  if (altHost) {
+    host = altHost;
+  }
+
+  const url = new URL(host);
+  url.pathname = STATS_ENDPOINT_PATH;
+
+  if (enrichedBidders.size === 0) {
+    return;
+  }
+
+  var payload = JSON.stringify({ bidders: [...enrichedBidders] });
+  if (navigator.sendBeacon !== undefined) {
+    navigator.sendBeacon(url.toString(), payload);
+    return;
+  }
+
+  if (window.fetch) {
+    fetch(url.toString(), {
```

Comment:
Please import communication methods from ajax.js

Reply:
Updated in 68e88b6